code/game/machinery/computer/chef_orders/chef_order.dm

```diff
@@ -0,0 +1,122 @@
+			order_datums += new path
+
+/obj/machinery/computer/chef_order/Destroy()
+	QDEL_NULL(radio)
+	. = ..()
+
+/obj/machinery/computer/chef_order/proc/get_total_cost()
+	. = 0
+	for(var/datum/orderable_item/item as anything in grocery_list)
+		for(var/i in 1 to grocery_list[item]) //for how many times we bought it
+			. += item.cost_per_order //add its price
+
+/obj/machinery/computer/chef_order/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "ProduceConsole", name)
+		ui.open()
+
+/obj/machinery/computer/chef_order/ui_data(mob/user)
+	. = ..()
+	.["off_cooldown"] = COOLDOWN_FINISHED(src, order_cooldown)
+
+/obj/machinery/computer/chef_order/ui_static_data(mob/user)
+	. = ..()
+	.["total_cost"] = get_total_cost()
+	.["order_datums"] = list()
+	for(var/datum/orderable_item/item as anything in order_datums)
+		.["order_datums"] += list(list(
+			"name" = item.name,
+			"desc" = item.desc,
+			"cat" = item.category_index,
+			"ref" = REF(item),
+			"cost" = item.cost_per_order,
+			"amt" = grocery_list[item]
+			))
+
+/obj/machinery/computer/chef_order/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	if(!isliving(usr))
+		return
+	var/mob/living/chef = usr
+	//this is null if the action doesn't need it (purchase, quickpurchase)
+	var/datum/orderable_item/wanted_item = locate(params["target"]) in order_datums
+	switch(action)
+		if("cart_set")
+			grocery_list[wanted_item] = params["amt"]
+			if(!grocery_list[wanted_item])
+				grocery_list -= wanted_item
+			update_static_data(chef)
+		if("purchase")
+			if(!grocery_list.len || !COOLDOWN_FINISHED(src, order_cooldown))
+				return
+			var/obj/item/card/id/chef_card = chef.get_idcard(TRUE)
+			if(!chef_card || !chef_card.registered_account)
+				say("No bank account detected!")
+				return
+			var/final_cost = get_total_cost()
+			if(!chef_card.registered_account.adjust_money(-final_cost))
+				say("Sorry, but you do not have enough money.")
+				return
+			say("Thank you for your purchase! It will arrive on the next cargo shuttle!")
+			var/message = "The kitchen has ordered groceries which will arrive on the cargo shuttle! Please make sure it gets to them as soon as possible!"
+			radio.talk_into(src, message, radio_channel)
+			COOLDOWN_START(src, order_cooldown, 60 SECONDS)
+			for(var/datum/orderable_item/ordered_item in grocery_list)
+				if(ordered_item in SSsupply.chef_groceries)
+					SSsupply.chef_groceries[ordered_item] += grocery_list[ordered_item]
+				else
+					SSsupply.chef_groceries[ordered_item] = grocery_list[ordered_item]
+			grocery_list.Cut()
+			update_static_data(chef)
+		if("express")
+			if(!grocery_list.len || !COOLDOWN_FINISHED(src, order_cooldown))
+				return
+			var/obj/item/card/id/chef_card = chef.get_idcard(TRUE)
+			if(!chef_card || !chef_card.registered_account)
+				say("No bank account detected!")
+				return
+			var/final_cost = get_total_cost()
+			final_cost *= 2
```

Comment:
This should use a formula to determine the express cost:

Minimun cost of 1.5x
Cost multiplier for express is otherwise equal to connected players divided by 10, which will make the multiplier scale up linearly with each player past 15.